samples/MQTTSN/src/paho_iot_hub_telemetry_example.c

```diff
@@ -0,0 +1,685 @@
+  char gateway_address[16];
+  int gateway_port;
+  char device_id[64];
+  unsigned short telemetry_topic_id;
+  az_iot_hub_client client;
+  short packet_id;
+} iothub_client_context;
+
+static void sleep_seconds(uint32_t seconds)
+{
+#ifdef _WIN32
+  Sleep((DWORD)seconds * 1000);
+#else
+  sleep(seconds);
+#endif
+}
+
+/*
+ * Read OS environment variables using stdlib function
+ */
+static az_result read_configuration_entry(
+    const char* name,
+    const char* env_name,
+    char* default_value,
+    bool hide_value,
+    az_span buffer,
+    az_span* out_value)
+{
+  printf("%s = ", name);
+  char* env = getenv(env_name);
+
+  if (env != NULL)
+  {
+    printf("%s\r\n", hide_value ? "***" : env);
+    az_span env_span = az_span_from_str(env);
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(buffer, az_span_size(env_span));
+    az_span_copy(buffer, env_span);
+    *out_value = az_span_slice(buffer, 0, az_span_size(env_span));
+  }
+  else if (default_value != NULL)
+  {
+    printf("%s\r\n", default_value);
+    az_span default_span = az_span_from_str(default_value);
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(buffer, az_span_size(default_span));
+    az_span_copy(buffer, default_span);
+    *out_value = az_span_slice(buffer, 0, az_span_size(default_span));
+  }
+  else
+  {
+    printf("(missing) Please set the %s environment variable.\r\n", env_name);
+    return AZ_ERROR_ARG;
+  }
+
+  return AZ_OK;
+}
+
+/*
+ * Read configurations and initialize Azure IoT Hub Client
+ */
+static int read_configuration_and_init_client(
+    az_iot_hub_client* client,
+    char* device_id,
+    int device_id_len,
+    char* gateway_address,
+    int gateway_address_len,
+    int* gateway_port,
+    char* iot_hub_hostname,
+    int iot_hub_hostname_len)
+{
+  // Read Device ID configuration
+  az_span device_id_span = az_span_init(device_id, device_id_len);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      ENV_DEVICE_ID, ENV_DEVICE_ID, "", false, device_id_span, &device_id_span));
+
+  // Read Gateway IP address configuration
+  az_span gateway_address_span = az_span_init(gateway_address, gateway_address_len);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      ENV_MQTTSN_GATEWAY_ADDRESS,
+      ENV_MQTTSN_GATEWAY_ADDRESS,
+      DEFAULT_GATEWAY_ADDRESS,
+      false,
+      gateway_address_span,
+      &gateway_address_span));
+
+  // Read IoT Hub Hostname configuration
+  az_span iot_hub_hostname_span = az_span_init(iot_hub_hostname, iot_hub_hostname_len);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      ENV_IOT_HUB_HOSTNAME,
+      ENV_IOT_HUB_HOSTNAME,
+      "",
+      false,
+      iot_hub_hostname_span,
+      &iot_hub_hostname_span));
+
+  // Initialize the hub client with the hub host endpoint and the default connection options
+  AZ_RETURN_IF_FAILED(az_iot_hub_client_init(
+      client,
+      az_span_slice(iot_hub_hostname_span, 0, (int32_t)strlen(iot_hub_hostname)),
+      az_span_slice(device_id_span, 0, (int32_t)strlen(device_id)),

```

Comment:
line 85 already does this for you.. you do need this ... =just use the span directly